beacon-chain/rpc/eth/events/events.go

```diff
@@ -750,7 +750,12 @@ func (s *Server) fillEventData(ctx context.Context, ev payloadattribute.EventDat
 // This event stream is intended to be used by builders and relays.
 // Parent fields are based on state at N_{current_slot}, while the rest of fields are based on state of N_{current_slot + 1}
 func (s *Server) payloadAttributesReader(ctx context.Context, ev payloadattribute.EventData) (lazyReader, error) {
-	ctx, cancel := context.WithTimeout(ctx, payloadAttributeTimeout)
+	deadline := ev.ProposalTime
```

Comment:
Is there an advantage to adding `ProposalTime` to `EventData` and passing it instead of calculating it locally? We already know the proposal time and are currently calculating it locally. 
For example:  
`payloadAttributeTimeout  := slots.ToTime(uint64(s.ChainInfoFetcher.GenesisTime().Unix()), ev.ProposalSlot)`

Reply:
I forgot we had access to gensis time via chain info and didn't want to hack a way to plumb it into the event service, but thanks for pointing that out, that's much better!